python/Ganga/Core/GangaRepository/SubJobXMLList.py

```diff
@@ -438,6 +441,9 @@ def flush(self):
                 subjob_data = self.__get_dataFile(str(index))
                 subjob_obj = self._cachedJobs[index]
 
+                if subjob_obj is subjob_obj._getRoot():
+                    raise GangaException(self, "Subjob parent not set correctly in flush.")
```

Comment:
This sounds like we should be adding some locks into this file as well for safety to do with loading of jobs.
I see how your change now means that we don't get conflicts from the job being in the cache before having it's parent set so this should avoid any future problems. I just worry about double-loading now which could cause issues.